### PR TITLE
refactor(core): organize resources package

### PR DESCRIPTION
**Pull request by hbradfield**
Wait for 2q release to GA before merging
https://github.ibm.com/IBMPrivateCloud/roadmap/issues/38284

### CODE REVIEW

**Comment by hbradfield** [comment]
ready for review

File: pkg/resources/configmaps.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	dataMap := make(map[string]string)
+	var err error
+	switch name {
+	case FluentdDaemonSetName + "-" + ConfigName:
+		dataMap[enableAuditLogForwardKey] = strconv.FormatBool(instance.Spec.Fluentd.EnableAuditLoggingForwarding)
+		type Data struct {
+			Value string `yaml:"fluent.conf"`
+		}
+		d := Data{}
+		err = yaml.Unmarshal([]byte(fluentdMainConfigData), &d)
```
Comment (lauriet-ibm) [inline pkg/resources/configmaps.go:185]: Should you be checking err, before setting the value?

File: pkg/resources/configmaps.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+		ds := DataS{}
+		var result string
+		if instance.Spec.Fluentd.JournalPath != "" {
+			result = sourceConfigData1 + instance.Spec.Fluentd.JournalPath + sourceConfigData2
+		} else {
+			result = sourceConfigData1 + defaultJournalPath + sourceConfigData2
+		}
+		p := strconv.Itoa(defaultHTTPPort)
+		result += sourceConfigData3 + p + sourceConfigData4
+		err = yaml.Unmarshal([]byte(result), &ds)
```
Comment (lauriet-ibm) [inline pkg/resources/configmaps.go:200]: Same comment here about checking err.

File: pkg/resources/configmaps.go [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+}
+
+func getConfig(data string) (string, error) {
+	var siemConfig string
+	// >([^]]+)< retrieves all data between match tags
+	regex := `>\n([^]]+)\n<`
+	reConfig := regexp.MustCompile(regex)
+	matches := reConfig.FindStringSubmatch(data)
+	if len(matches) < 2 {
+		return "", errors.New("output plugin config misformatted")
```
Comment (lauriet-ibm) [inline pkg/resources/configmaps.go:238]: Do you want to add the incorrect value to the log output?

File: pkg/resources/configmaps.go (hunk 1 above)
Comment (hbradfield) [reply pkg/resources/configmaps.go:185]: Unmarshal returns an error https://godoc.org/gopkg.in/yaml.v2#Unmarshal

File: pkg/resources/configmaps.go (hunk 2 above)
Comment (hbradfield) [reply pkg/resources/configmaps.go:200]: Unmarshal returns an error https://godoc.org/gopkg.in/yaml.v2#Unmarshal

File: pkg/resources/configmaps.go (hunk 1 above)
Comment (lauriet-ibm) [reply pkg/resources/configmaps.go:185]: Let me ask a different way, If you get an err, at 185, should you test the err, before setting the value at 186?

Comment (hbradfield) [reply pkg/resources/configmaps.go:185]: I see what you mean, I added breaks before the key, value assignment in the map if there is an error

File: pkg/resources/configmaps.go (hunk 3 above)
Comment (hbradfield) [reply pkg/resources/configmaps.go:247]: updated

**Pull request merged**